huaweicloud/services/waf/data_source_huaweicloud_waf_certificate.go

```diff
@@ -0,0 +1,85 @@
+package waf
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/huaweicloud/golangsdk/openstack/waf/v1/certificates"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+)
+
+const (
+	// EXP_STATUS_NOT_EXPIRED not expired
+	EXP_STATUS_NOT_EXPIRED = 0
+	// EXP_STATUS_EXPIRED has expired
+	EXP_STATUS_EXPIRED = 1
+	// EXP_STATUS_EXPIRED_SOON will expire soon
+	EXP_STATUS_EXPIRED_SOON = 2
+)
+
+func DataSourceWafCertificateV1() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceWafCertificateV1Read,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"expire_status": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  EXP_STATUS_NOT_EXPIRED,
+				ValidateFunc: validation.IntInSlice([]int{
+					EXP_STATUS_NOT_EXPIRED, EXP_STATUS_EXPIRED, EXP_STATUS_EXPIRED_SOON,
+				}),
+			},
+		},
+	}
+}
+
+func dataSourceWafCertificateV1Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	wafClient, err := config.WafV1Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.Errorf("error creating HuaweiCloud WAF Client: %s", err)
+	}
+
+	expStatus := d.Get("expire_status").(int)
+	listOpts := certificates.ListOpts{
+		Page:      1,
+		Pagesize:  5,
```

Comment:
what does it mean when Page = 1 and Pagesize =5 ?

Reply:
query the first 5 records only